discv4/kademlia/table.go

```diff
@@ -17,10 +18,20 @@ const (
 	minLogDistance = addrByteSize + 1 - bucketsCount
 )
 
+func New() *Table {
+	t := new(Table)
+	for i := 0; i < bucketsCount; i++ {
+		t.buckets[i] = &kBucket{
+			lru:     list.List{},
+			entries: map[eth.Hash]*list.Element{},
+		}
+	}
+	return t
+}
+
 type Table struct {
-	mu       sync.Mutex // protects buckets
-	selfNode *enr.Record
```

Comment:
I feel pretty strongly that `selfNode` should be an "instance var" here. A Kademlia table, by definition and design, is only supposed to work with a single node as the reference point. Changing the API of `Insert` to take in an arbitrary `self` ENR record is exposing the package to misuse.

Reply:
That makes sense. I'll set it up that way.

Reply:
added!